securityscanutils/securityscan.go

```diff
@@ -233,23 +259,29 @@ func (r *SecurityScanRepo) RunMarkdownScan(ctx context.Context, release *github.
 			imageWithRepo = fmt.Sprintf("%s/%s:%s", r.Opts.ImageRepo, image, version)
 		}
 		fileName := fmt.Sprintf("%s_cve_report.docgen", image)
-		output := path.Join(outputDir, fileName)
+		output := path.Join(trivyScanOutputDir, fileName)
 		_, vulnFound, err := r.trivyScanner.ScanImage(ctx, imageWithRepo, markdownTplFile, output)
 		if err != nil {
-			return eris.Wrapf(err, "error running image scan on image %s", imageWithRepo)
+			if errors.Is(err, UnrecoverableErr) {
+				return eris.Wrapf(err, "error running image scan on image %s", imageWithRepo)
+			}
+			vulnerabilityMd += fmt.Sprintf("# %s\n\n %s", imageWithRepo, err)
 		}
 
 		if vulnFound {
-			trivyScanMd, err := ioutil.ReadFile(output)
+			trivyScanMd, err := os.ReadFile(output)
 			if err != nil {
 				return eris.Wrapf(err, "error reading trivy markdown scan file %s to generate github issue", output)
 			}
 			vulnerabilityMd += fmt.Sprintf("# %s\n\n %s\n\n", imageWithRepo, trivyScanMd)
 		}
 
 	}
-	// Create / Update Github issue for the repo if a vulnerability is found
-	return r.githubIssueWriter.CreateUpdateVulnerabilityIssue(ctx, release, vulnerabilityMd)
+	if vulnerabilityMd != "" && r.Opts.AdditionalContext != "" {
+		vulnerabilityMd = fmt.Sprintf("%s\n%s", r.Opts.AdditionalContext, vulnerabilityMd)
+	}
```

Comment:
I like this! Perhaps just a clarifying question: is the additional context intended to just be used for the localWriter, or should it apply to all implementations?

Reply:
The intention would be to apply to all implementations